purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/Localization.kt

```diff
@@ -0,0 +1,58 @@
+package com.revenuecat.purchases.paywalls.components
+
+import com.revenuecat.purchases.paywalls.components.properties.ThemeImageUrls
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.InternalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.PolymorphicKind
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+@Serializable
+@JvmInline
+internal value class LocaleId(val value: String)
+
+internal typealias LocalizationKey = String
+
+internal typealias LocalizationDictionary = Map<LocalizationKey, LocalizationData>
+
+/**
+ * A make-shift union type. LocalizationData is either a plain String or a ThemeImageUrls object.
+ */
+@Serializable(with = LocalizationDataSerializer::class)
+internal sealed interface LocalizationData {
+    @Serializable
+    @JvmInline
+    value class Text(val value: String) : LocalizationData
+
+    @Serializable
+    @JvmInline
+    value class Image(val value: ThemeImageUrls) : LocalizationData
+}
+
+private object LocalizationDataSerializer : KSerializer<LocalizationData> {
+    // Documentation says to use either PrimitiveSerialDescriptor or buildClassSerialDescriptor. However, we need a
+    // polymorphic descriptor that's either a primitive (string) or a class. So neither of those options fit the bill.
+    @OptIn(InternalSerializationApi::class, ExperimentalSerializationApi::class)
+    override val descriptor: SerialDescriptor = buildSerialDescriptor(
+        serialName = "LocalizationData",
+        kind = PolymorphicKind.SEALED,
+    )
```

Comment:
I get the feeling that kotlinx.serialization doesn't quite "agree" with what we're doing, which is the same JSON key sometimes pointing to a primitive (string) and sometimes to a class (`ThemeImageUrls`). That's why I had to add these annotations to make it work.

Reply:
Ahhhh yeaaah, I can see this not working well in some places 😬 It felt surpringly well in Swift but 🤔 I can't see us adding more than just text or images though. Do you think this is still okay to go this way?

Reply:
As discussed, we'll explore changing the schema for this. I'm going to merge this now and will update the SDK separately.